be/src/formats/parquet/scalar_column_reader.cpp

```diff
@@ -342,6 +351,84 @@ StatusOr<bool> RawColumnReader::_page_index_zone_map_filter(const std::vector<co
     return true;
 }
 
+Status RawColumnReader::_init_column_bloom_filter(int offset, int length, BloomFilter& bloom_filter) const {
+    std::vector<char> bloom_filter_data;
+    tparquet::BloomFilterHeader header;
+    uint32_t header_len = SBBF_HEADER_SIZE_ESTIMATE;
+    if (length > 0) {
+        bloom_filter_data.resize(length + 1);
+        RETURN_IF_ERROR(_opts.file->read_at_fully(offset, bloom_filter_data.data(), length));
+    } else {
+        // if length is not set, read the header first;
+        bloom_filter_data.reserve(header_len);
+        RETURN_IF_ERROR(_opts.file->read_at_fully(offset, bloom_filter_data.data(), header_len));
+    }
+
+    RETURN_IF_ERROR(deserialize_thrift_msg(reinterpret_cast<const uint8*>(bloom_filter_data.data()), &header_len,
+                                           TProtocolType::COMPACT, &header));
+    if (length == 0) {
+        bloom_filter_data.resize(header_len + header.numBytes + 1);
+        RETURN_IF_ERROR(
+                _opts.file->read_at_fully(offset + header_len, bloom_filter_data.data() + header_len, header.numBytes));
+    }
+    if (get_chunk_metadata()->meta_data.__isset.statistics &&
+        get_chunk_metadata()->meta_data.statistics.__isset.null_count) {
+        if (get_chunk_metadata()->meta_data.statistics.null_count > 0) {
+            bloom_filter_data.back() = 1;
+        } else {
+            bloom_filter_data.back() = 0;
+        }
+    } else if (get_column_parquet_field()->is_nullable) {
+        bloom_filter_data.back() = 1; //set has null as default, to avoid `column is null` to filter the group.
+    } else {
+        bloom_filter_data.back() = 0;
+    }
+
+    return bloom_filter.init(bloom_filter_data.data() + header_len, header.numBytes + 1, HashStrategyPB::XXHASH64);
+}
+
+StatusOr<bool> RawColumnReader::_row_group_bloom_filter(const std::vector<const ColumnPredicate*>& predicates,
+                                                        CompoundNodeType pred_relation, const TypeDescriptor& col_type,
+                                                        const uint64_t rg_first_row, const uint64_t rg_num_rows) const {
+    bool filtered = false;
+    auto& column_metadata = get_chunk_metadata()->meta_data;
+    bool has_bloom_filter = column_metadata.__isset.bloom_filter_offset;
+    if (!has_bloom_filter) {
+        // bloom filter is not existed, select all
+        return filtered = false;
+    } else if (get_column_parquet_field() == nullptr) {
+        // which scenerio will reach here? is it unexpected?
```

Comment:
Columns like `FixedValueColumnReader` will not have Parquet related metadata and fields